PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/MySQLConnectionUtils.java

```diff
@@ -27,49 +26,54 @@ public static boolean connect(String url, String username, String password) {
             Class.forName("com.mysql.jdbc.Driver");
             DriverManager.getConnection(url, username, password);
             return true;
-        } catch (ClassNotFoundException | SQLException exception) {
+        } catch (final ClassNotFoundException | SQLException exception) {
+            return false;
         }
-        return false;
     }
 
     public static ConnectResult connectWithPing(String url, String username, String password) {
+        int errorCode = 0;
         boolean connected = false;
         String errorMessage = null;
         Long pingCost = null;
         String serverVersion = null;
         // refresh property
         try {
             Class.forName("com.mysql.jdbc.Driver");
-            long start = System.currentTimeMillis();
-            Connection connection = DriverManager.getConnection(url, username, password);
+            final long start = System.currentTimeMillis();
+            final Connection connection = DriverManager.getConnection(url, username, password);
             connected = true;
-            Statement statement = connection.createStatement();
-            ResultSet resultSet = statement.executeQuery("select 'hi'");
+            final Statement statement = connection.createStatement();
+            final ResultSet resultSet = statement.executeQuery("select 'hi'");
             if (resultSet.next()) {
-                String result = resultSet.getString(1);
+                final String result = resultSet.getString(1);
                 connected = "hi".equals(result);
             }
             pingCost = System.currentTimeMillis() - start;
             serverVersion = ((ConnectionImpl) connection).getServerVersion().toString();
-        } catch (ClassNotFoundException | SQLException exception) {
+        } catch (final ClassNotFoundException exception) {
+            errorMessage = exception.getMessage();
```

Comment:
errorCode = -1? it is strange to have error code == 0 and non-blank error message